### ISSUE TITLE
Bug: Price is not kept in purchase form

### ISSUE THREAD

**Issue by dabenzel** [grocy/grocy#1917]
When I enter a price in the purchase-form and press TAB, the price gets deleted from the field and in the console there is a js-error.

Example:
![msedge_wIPBBtcFmk](https://user-images.githubusercontent.com/2480235/173184665-6792b77f-c6ac-4a7a-a273-713abdddfb3c.gif)

Stack-Trace of the error:
```
  | (anonym) | @ | jquery.min.js?v=3.3.1:2
-- | -- | -- | --
  | each | @ | jquery.min.js?v=3.3.1:2
  | each | @ | jquery.min.js?v=3.3.1:2
  | val | @ | jquery.min.js?v=3.3.1:2
  | (anonym) | @ | numberpicker.js?v=3.3.1:109
  | dispatch | @ | jquery.min.js?v=3.3.1:2
  | v.handle | @ | jquery.min.js?v=3.3.1:2

```

I'm using Grocy 3.3.1 from [linuxserver/grocy](https://hub.docker.com/r/linuxserver/grocy/) tag: latest behind a reverse-Proxy (nginx)

**Comment by berrnd** [comment]
Needs a reproducible example as mentioned on the issue template and also just in #1915 - I can't fix things without having something reproducible. This can be reopened when there is such an reproducible example.

Not reproducible on the current demo:

https://user-images.githubusercontent.com/2161815/173184154-1464dd7a-d2e0-427f-8839-e8241faa0add.mp4

**Comment by dabenzel** [comment]
was able to reproduce it in demo.

When you tick this setting:
![image](https://user-images.githubusercontent.com/2480235/173185139-b9e508b8-df17-418d-ba3b-5655c770ef1d.png)

in `/stocksettings` the bug is also in the demo.

When I untick it on my system, the price doesn't get deleted either

**Comment by berrnd** [comment]
Well, why enabling this setting when you want to input the decimal separator manually?

Of course a fix to prevent this can be added, but it's just [GIGO](https://en.wikipedia.org/wiki/Garbage_in,_garbage_out) at the end - grocy is generally based on this.

This setting was introduced by #1345, someone wanted to be able to input prices like on a Supermarktkasse.

**Comment by dabenzel** [comment]
It also doesn't work when you input a price like `399` for `3,99`

**Comment by berrnd** [comment]
> It also doesn't work when you input a price like `399` for `3,99`

Of course it does, that's what this setting was built for:

![grafik](https://user-images.githubusercontent.com/2161815/173185542-dc51030d-108a-47b4-a326-51f2afe23ec6.png)

https://user-images.githubusercontent.com/2161815/173185653-6cc115d5-e57b-49a4-8828-4afff47fa0a1.mp4

Otherwise please (just again): Provide a reproducible example. Mabye it depends on your browser, your locale setting, or whatever.

**Comment by dabenzel** [comment]
Browser is Microsoft Edge Chromium
Version 102.0.1245.39 (Offizielles Build) (64-Bit)

Reproduced it in demo (inPrivate-Window with no addons enabled):


https://user-images.githubusercontent.com/2480235/173186032-64ae45b2-e2cd-4e91-ad7b-6712d52dd10f.mp4

**Comment by berrnd** [comment]
> Browser is Microsoft Edge Chromium

Is fixed by dbf660f953e590022f0a44372c9468423c8967cb.
